src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/ConvertDataTemplateDirectoryProviderFunctionalTests.cs

```diff
@@ -0,0 +1,128 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using DotLiquid;
+using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
+using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Microsoft.Health.Fhir.Liquid.Converter.Models.Hl7v2;
+using Microsoft.Health.Fhir.Liquid.Converter.Processors;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests
+{
+    public class ConvertDataTemplateDirectoryProviderFunctionalTests : BaseConvertDataFunctionalTests
```

Comment:
Not sure I understand how the new categories are being applied.  For example why is the test "GivenCcdaMessageForTimezoneTesting_WhenConvert_ExpectedResultShouldBeReturned" in this file now?

Reply:
I basically wanted to reuse a lot of the tests in the original FunctionalTests.cs file (which was previously testing against template directory) to also test using template collection provider. 
So, this file corresponds to the previous FunctionalTests.cs and the new base class has the copied common functions that can be used across both.
Specifically, the test you mentioned was always in this test class, and I didn't take it over as a common test since it seemed like the logic being tested is agnostic to whether template directory or template collection is used.

Reply:
Okay, my follow up question is around having functional tests per template type, like CCDA tests specific to the functionality being tested in the test case.  It seems like we would benefit from getting more specific in the long run.

Reply:
There are tests per template type in these tests, are you looking for something specific? Those are the common ones I pulled out into the base class so we can test when using both - a TemplateCollectionProvider as well as TemplateDirectoryProvider.

Reply:
Sorry for not being more clear, I was asking if we should have dedicated test files/directories for the different template types rather than combining them into one file.  I don't think it should impact this PR.  We can do a subsequent update if that is change we decide to make.

Reply:
Gotcha, yeah agreed, I've also noticed the need for similar refactoring to improve these tests (and others in this repo) generally, that we can chip away at as and when we make changes to the relevant classes or as part of backlog. For now, given I was making changes to the template collection provider, I wanted to get test coverage on that so added in these tests (and given this is already a large PR didn't want to include more changes yet), but yes incrementally we should continue to improve the organization of these tests.